zio-http-benchmarks/src/main/scala/zio/http/benchmark/RoutesBenchmark.scala

```diff
@@ -0,0 +1,67 @@
+package zio.http.benchmark
+
+import java.util.concurrent.TimeUnit
+
+import scala.util.Random
+
+import zio.{Runtime, Unsafe, ZIO}
+
+import zio.http.endpoint.Endpoint
+import zio.http.{Handler, Method, Request, Routes}
+
+import org.openjdk.jmh.annotations._
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class RoutesBenchmark {
+
+  val REPEAT_N = 1000
+
+  val paths = ('a' to 'z').inits.map(_.mkString).toList.reverse.tail
+
+  val routes = Routes.fromIterable(paths.map(p => Endpoint(Method.GET / p).out[Unit].implementHandler(Handler.unit)))
+
+  val requests = paths.map(p => Request.get(p))
+
+  def request: Request = requests(Random.nextInt(requests.size))
+
+  def unsafeRun[E, A](zio: ZIO[Any, E, A]): Unit = Unsafe.unsafe { implicit unsafe =>
+    Runtime.default.unsafe
+      .run(zio.unit)
+      .getOrThrowFiberFailure()
+  }
+
+  val paths2 = ('b' to 'z').inits.map(_.mkString).toList.reverse.tail
+
+  val routes2 = Routes.fromIterable(paths2.map(p => Endpoint(Method.GET / p).out[Unit].implementHandler(Handler.unit)))
+
+  val requests2 = requests ++ paths2.map(p => Request.get(p))
+
+  def request2: Request = requests2(Random.nextInt(requests2.size))
+
+  @Benchmark
+  def benchmarkSmallDataZioApi(): Unit =
+    unsafeRun {
+      ZIO.collectAllDiscard(1.to(REPEAT_N).map(_ => routes.run(request)))
```

Comment:
Might be better to use `routes.isDefinedAt` to avoid the noise added by running the ZIO effects